node/node.go

```diff
@@ -0,0 +1,185 @@
+package node
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/url"
+	"os"
+	"path/filepath"
+	"slices"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nkeys"
+)
+
+type Node interface {
+	Validate() error
+	Start()
+}
+
+type nexNode struct {
+	nc *nats.Conn
+
+	logger                *slog.Logger
+	agentHandshakeTimeout int
+	resourceDirectory     string
+	internalNodeNATHost   string
+	internalNodeNATPort   int
+	microVMMode           bool
```

Comment:
Does the  node care about this anymore since the agent now decides whether to use firecracker?

Maybe defaultMicrovmMode or similar

Reply:
it cares about it for preflight/valuations....unless we are going to move that to the agents (no please).  As it sits now, we don't check or install for the firecracker/microvm dependencies unless we tell it to.  

To your point, I can see why this wouldn't be the nodes problem anymore.  That begs the question, whose job is it to check dependencies.  Example I just though of...if one day we have a "python agent" that runs scripts...whos job is it going to be to ensure the `python` binary exists?